### PR TITLE
Add --project option to man page

### LINKED ISSUE #27938: document command line flag --project
Eg in `julia --help`, and also the manual. At least that the flag exists and sets the project, the concept of the project is (will be?) explained in the relevant section of the manual (maybe cross-reference).

cf 955a950342ebba88d05aa1b82553500a283ecb60, #27816

### PR DESCRIPTION
**Pull request by simonbyrne**
Fixes #27938.

### CODE REVIEW

File: doc/man/julia.1 [hunk 1]
```diff
@@ -63,6 +63,12 @@ Display version information
 -h, --help
 Print help message
 
+.TP
+--project=<dir>/@.
+Activate the package environment in the given directory. The @. option will
```
Comment (fredrikekre) [inline doc/man/julia.1]: Maybe "Set dir as the home project/environment" to differentiate this from actually activating with `Pkg.activate`?

Comment (simonbyrne) [reply doc/man/julia.1]: That's a better idea, thanks!

**Review by fredrikekre — APPROVED** [review]

**Comment by KristofferC** [comment]
Ref https://github.com/JuliaLang/julia/pull/29191 which also mentioned the default when only `--project` is given.

**Comment by simonbyrne** [comment]
Thanks: see #29731 for the fix.

**Pull request merged**